lib/Resource.js

```diff
@@ -330,6 +330,13 @@ class Resource {
 
 	/**
 	 * Retrieve the project assigned to the resource
+	 * <br/>
+	 * <b>Note for UI5 Tooling extensions (i.e. custom tasks, custom middleware):</b>
+	 * In order to ensure compatibility across UI5 Tooling versions, consider using the
+	 * <code>getProject(resource)</code> methods provided by
+	 * [TaskUtil]{@link module:@ui5/project/build/helpers/TaskUtil} and
+	 * [MiddlewareUtil]{@link module:@ui5/server.middleware.MiddlewareUtil}, which will
+	 * return a Specification Version-compatible Project interface.
```

Comment:
in other PRs you have used the term `Specification Version-dependent`. IMHO we should use only one of the terms

Reply:
@KlattG any preferred variant?

Reply:
@flovogt I have no problem with using both terms. 'compatible' sounds ideal here.